<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DetectFoc</class>
 <widget class="QWidget" name="DetectFoc">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>791</width>
    <height>196</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Detect and Calculate Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0,0,0,0">
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>3</number>
      </property>
      <property name="rightMargin">
       <number>3</number>
      </property>
      <property name="bottomMargin">
       <number>3</number>
      </property>
      <property name="spacing">
       <number>3</number>
      </property>
      <item row="4" column="6">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>← Calc</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="2">
       <widget class="QDoubleSpinBox" name="resistanceBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Motor resistance</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>R: </string>
        </property>
        <property name="suffix">
         <string> mΩ</string>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="maximum">
         <double>100000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.001000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="6">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>← Calc</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2" colspan="2">
       <widget class="QDoubleSpinBox" name="obsGainBox">
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Observer gain</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>Observer Gain (x1M): </string>
        </property>
        <property name="suffix">
         <string/>
        </property>
        <property name="maximum">
         <double>10000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="6">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>← Param</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QDoubleSpinBox" name="inductanceBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Motor inductance</string>
        </property>
        <property name="wrapping">
         <bool>false</bool>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>L: </string>
        </property>
        <property name="suffix">
         <string> µH</string>
        </property>
        <property name="maximum">
         <double>100000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QDoubleSpinBox" name="currentBox">
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Current for starting</string>
        </property>
        <property name="prefix">
         <string>I: </string>
        </property>
        <property name="suffix">
         <string> A</string>
        </property>
        <property name="maximum">
         <double>100000.000000000000000</double>
        </property>
        <property name="value">
         <double>5.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="calcGainButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Calculate Observer Gain (requires λ)</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../res.qrc">
          <normaloff>:/res/icons/Calculator-96.png</normaloff>:/res/icons/Calculator-96.png</iconset>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="calcKpKiButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Calculate KP and KI (requires R and L)</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../res.qrc">
          <normaloff>:/res/icons/Calculator-96.png</normaloff>:/res/icons/Calculator-96.png</iconset>
        </property>
       </widget>
      </item>
      <item row="1" column="6">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>← For λ</string>
        </property>
       </widget>
      </item>
      <item row="2" column="5">
       <widget class="QDoubleSpinBox" name="lambdaBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Motor flux linkage</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>λ: </string>
        </property>
        <property name="suffix">
         <string> mWb</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.000100000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="7">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="helpButton">
          <property name="toolTip">
           <string>Show help</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../res.qrc">
            <normaloff>:/res/icons/Help-96.png</normaloff>:/res/icons/Help-96.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../res.qrc">
            <normaloff>:/res/icons/arrow_r.png</normaloff>
            <disabledoff>:/res/icons/arrow_r.png</disabledoff>
            <disabledon>:/res/icons/arrow_r.png</disabledon>:/res/icons/arrow_r.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>45</width>
            <height>16</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="rlButton">
          <property name="toolTip">
           <string>Measure resistance and inductance</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../res.qrc">
            <normaloff>:/res/icons/rl.png</normaloff>:/res/icons/rl.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>16</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../res.qrc">
            <normaloff>:/res/icons/arrow_r.png</normaloff>
            <disabledoff>:/res/icons/arrow_r.png</disabledoff>
            <disabledon>:/res/icons/arrow_r.png</disabledon>:/res/icons/arrow_r.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>45</width>
            <height>16</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="lambdaButton">
          <property name="toolTip">
           <string>Measure flux linkage</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../res.qrc">
            <normaloff>:/res/icons/lambda.png</normaloff>:/res/icons/lambda.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>16</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_3">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../res.qrc">
            <normaloff>:/res/icons/arrow_r.png</normaloff>
            <disabledoff>:/res/icons/arrow_r.png</disabledoff>
            <disabledon>:/res/icons/arrow_r.png</disabledon>:/res/icons/arrow_r.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>45</width>
            <height>16</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="applyAllButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Calculate and apply all parameters</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../res.qrc">
            <normaloff>:/res/icons/apply.png</normaloff>:/res/icons/apply.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>50</width>
            <height>16</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="3">
       <widget class="QDoubleSpinBox" name="dutyBox">
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Duty cycle to sample at</string>
        </property>
        <property name="prefix">
         <string>D: </string>
        </property>
        <property name="suffix">
         <string/>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.300000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QDoubleSpinBox" name="inductanceDiffBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>Lq-Ld: </string>
        </property>
        <property name="suffix">
         <string> µH</string>
        </property>
        <property name="maximum">
         <double>999999.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="4" colspan="2">
       <widget class="QDoubleSpinBox" name="erpmBox">
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;ERPM/sec to ramp up the motor in openloop mode with.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="prefix">
         <string>α: </string>
        </property>
        <property name="suffix">
         <string> ERPM/s</string>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="maximum">
         <double>100000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>10.000000000000000</double>
        </property>
        <property name="value">
         <double>2000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="4" colspan="2">
       <widget class="QPushButton" name="calcApplyLocalButton">
        <property name="toolTip">
         <string>Calculate KP, KI and Observer Gain from the entered settings and the time constant, and apply them.</string>
        </property>
        <property name="text">
         <string>Calc Apply Old</string>
        </property>
        <property name="icon">
         <iconset resource="../res.qrc">
          <normaloff>:/res/icons/Calculator-96.png</normaloff>:/res/icons/Calculator-96.png</iconset>
        </property>
       </widget>
      </item>
      <item row="3" column="5">
       <widget class="QDoubleSpinBox" name="kiBox">
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Current control KI</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>KI: </string>
        </property>
        <property name="suffix">
         <string/>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="maximum">
         <double>100000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="4">
       <widget class="QDoubleSpinBox" name="kpBox">
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Current control KP</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>KP: </string>
        </property>
        <property name="suffix">
         <string/>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="2" colspan="2">
       <widget class="QDoubleSpinBox" name="tcBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>DejaVu Sans Mono</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Time constant for calculating KP and KI</string>
        </property>
        <property name="prefix">
         <string>T: </string>
        </property>
        <property name="suffix">
         <string> µS</string>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="maximum">
         <double>100000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>100.000000000000000</double>
        </property>
        <property name="value">
         <double>1000.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../res.qrc"/>
 </resources>
 <connections/>
</ui>
